app/controllers/api/oceans_controller.rb

```diff
@@ -1,11 +1,15 @@
 class Api::OceansController < ApplicationController
   def index
     params = Rack::Utils.parse_query URI(request.url).query
-    n_oceans = params['n_oceans']
-    n_oceans = '1' if n_oceans.empty?
-    oceans = Array.new(n_oceans.to_i) do
-      OceanService.new(Random.rand(1..5)).call
+    count = params['count']
+    count = '4' if count.nil? || count.empty?
+    if count.to_i < 0 || count.to_i > 20
+      render :nothing => true, :status => :bad_request  
+    else
```

Comment:
it would be great if you would add private methods to the controller to get the param value like this: 

```
def number_of_oceans
  @number_of_oceans ||= params.fetch(:limit, 4).to_i
end

def render_bad_request
  render nothing: true, status: :bad_request
end

def limit_valid?
  number_of_oceans.between?(0, 20)
end
```

and then add a before filter to the top of the class

```
before_filter :render_bad_request, unless: limit_valid?
```

do in the index method you just need to:

```
def index
  oceans = Array.new(number_of_oceans) do
    OceanService.new(Random.rand(1..5)).call
  end
  render json: oceans
end
```

Like this, the controller is way more readable, with the methods being in place